janusgraph-hadoop-parent/janusgraph-hadoop-core/src/main/java/org/janusgraph/hadoop/formats/cassandra/CassandraBinaryInputFormat.java

```diff
@@ -65,10 +65,14 @@
     @Override
     public RecordReader<StaticBuffer, Iterable<Entry>> createRecordReader(final InputSplit inputSplit, final TaskAttemptContext taskAttemptContext)
             throws IOException, InterruptedException {
-        columnFamilyRecordReader =
-                (ColumnFamilyRecordReader)columnFamilyInputFormat.createRecordReader(inputSplit, taskAttemptContext);
-        janusgraphRecordReader =
+        if (Boolean.getBoolean("use.cassandra.3")) { // see issue 172
+            janusgraphRecordReader = new CqlBridgeRecordReader();
+        } else { // the default case, Cassandra 2.1.9
+            columnFamilyRecordReader =
+                (ColumnFamilyRecordReader) columnFamilyInputFormat.createRecordReader(inputSplit, taskAttemptContext);
```

Comment:
Are you asking why the _original_ code is casting it? It's perhaps BigTable-inspired technologies (Cassandra is an instance) use a term `ColumnFamily` that is akin to a relational `Table`. In any case, I left it behind so that I don't cause any unintended backward incompatibility in the _new_ code.